src/core/lib/event_engine/windows/windows_endpoint.cc

```diff
@@ -356,7 +355,7 @@ void WindowsEndpoint::HandleWriteClosure::Run() {
   } else {
     GPR_ASSERT(result.bytes_transferred == buffer_->Length());
   }
-  Reset();
+  std::ignore = ResetAndReturnCallback();
```

Comment:
Not really familiar with this code, would removing line 350 and 359 and just use `return ResetAndReturnCallback()(status);` here have the same effect?